src/roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -139,24 +150,33 @@ def make_nwb_metadata(self, nwbfile, opts):
             ),
         )
 
-    @check_get_frames_args
-    def get_frames(self, frame_idxs, channel=0):
-        opts = self.nwbfile.acquisition[self._optical_series_name]
-        if frame_idxs.size > 1 and np.all(np.diff(frame_idxs) > 0):
-            return opts.data[frame_idxs].transpose([0, 2, 1])
+    def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0):
+
+        # Fancy indexing is non performant for h5.py with long frame lists
+        if frame_idxs is not None:
+            slice_start = np.min(frame_idxs)
+            slice_stop = min(np.max(frame_idxs) + 1, self.get_num_frames())
         else:
-            sorted_idxs = np.sort(frame_idxs)
-            argsorted_idxs = np.argsort(frame_idxs)
-            return opts.data[sorted_idxs][argsorted_idxs].transpose([0, 2, 1])
-
-    @check_get_videos_args
-    def get_video(self, start_frame=None, end_frame=None, channel=0):
-        opts = self.nwbfile.acquisition[self._optical_series_name]
-        video = opts.data[start_frame:end_frame].transpose([0, 2, 1])
```

Comment:
Turns out this un-transposition is actually correct - can you adjust that and the tests and then we can call this good?